### PR TITLE
ALC-300 swap history tab

### PR DESCRIPTION
**Pull request by hosainnet**
Display swap history tab

Screenshot 2021-05-05 at 18 14 11

- [x] Branch name corresponds to Linear ticket number.
- [x] Added sufficient integration and unit tests.
- [x] Addressed a single concern in the least number of changed lines as
      possible.
- [x] Only fixed/added the functionality in question **OR** address wide-spread
      whitespace/style issues, not both.

Thanks!

### CODE REVIEW

File: src/components/RightPanel/index.tsx [hunk 1]
```diff
@@ -1,57 +1,31 @@
...
-
-  if (graphqlError) return <p>{graphqlError}</p>;
-  if (!crpData) return <p>Loading...</p>;
-
-  const poolDetail = mapPoolResponseToPoolDetail(
-    poolDetails as PoolApiEntry,
-    crpData,
-    tokenPrices
-  );
-
```
Comment (hosainnet) [inline src/components/RightPanel/index.tsx:45]: Moved this fetching logic to parent component and pass props in

File: src/pages/PoolDashboard/index.tsx [hunk 2]
```diff
@@ -63,33 +68,81 @@ const PoolDashboard = ({ poolDetails }: PoolDashboardProps) => {
...
+
+  useEffect(() => {
+    if (poolSwapsResponse && tokenPricesResponse) {
+      setPoolSwaps(mapSwaps(poolSwapsResponse.swaps, tokenPricesResponse));
+    }
+  }, [poolSwapsResponse, tokenPricesResponse]);
+
+  /**
+   * Fetch pool metadata
+   */
```
Comment (hosainnet) [inline src/pages/PoolDashboard/index.tsx:90]: Moved from above

File: src/pages/PoolDashboard/index.tsx [hunk 3]
```diff
@@ -63,33 +68,81 @@ const PoolDashboard = ({ poolDetails }: PoolDashboardProps) => {
       });
   }, []);
 
-  if (loading || loadingPricesLoading) {
+  /**
+   * Fetch pool swaps
+   */
```
Comment (hosainnet) [inline src/pages/PoolDashboard/index.tsx:73]: This is where we fetch the swap history

File: src/utils/metrics.ts [hunk 4]
```diff
@@ -39,17 +40,3 @@ export const processPoolMetrics = (
...
-  prices: TokenPrices
-): number {
-  if (tokenAddress === constants.TOKEN_SYMBOLS_TO_ADDRESS["WETH"]) {
-    return (
-      amount *
-      parseInt(prices[constants.TOKEN_SYMBOLS_TO_ADDRESS["WETH"]].price)
-    );
-  }
-  return amount;
-}
```
Comment (hosainnet) [inline src/utils/metrics.ts:55]: Moved to price util to be re-used

File: src/queries/swap.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { gql } from "@apollo/client";
+
+export const GET_SWAPS_BY_POOL_ADDRESS = gql`
+  query($poolAddress: String) {
+    swaps(
+      first: 100
+      orderBy: timestamp
+      orderByDirection: desc
```
Comment (jamespfarrell) [inline src/queries/swap.ts]: I believe the api syntax is:
```
orderDirection: desc
```
not 
```
orderByDirection: desc
```

**Review by jamespfarrell — CHANGES_REQUESTED** [review]

File: src/components/SwapHistory/index.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+          <Th>Swap Fee</Th>
+          <Th>Wallet</Th>
+        </Tr>
+        {swapHistory.map((swap) => {
+          let date = DateTime.fromSeconds(swap.timestamp).toFormat("ff");
+          return (
+            <Tr key={swap.id}>
+              <Td>
+                <Link
+                  href={`https://etherscan.io/tx/${getTxHashFromId(swap.id)}`}
```
Comment (ledes) [inline src/components/SwapHistory/index.tsx:33]: minor but you can have a const with `https://etherscan.io/` and reuse it

**Review by ledes — DISMISSED** [review]

File: src/queries/swap.ts (hunk 5 above)
Comment (hosainnet) [reply src/queries/swap.ts]: Thanks 👍

File: src/components/SwapHistory/index.tsx (hunk 6 above)
Comment (hosainnet) [reply src/components/SwapHistory/index.tsx]: 👍

File: src/components/SwapHistory/index.tsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+import { formatNumber } from "utils/number";
+import { BiLinkExternal } from "react-icons/bi";
+import constants from "../../constants";
+
+export type SwapProps = {
+  swapHistory: Swap[];
+  launchToken: string;
+};
+
+export default function Index({ swapHistory, launchToken }: SwapProps) {
```
Comment (ledes) [inline src/components/SwapHistory/index.tsx:15]: you want to name the function like the name of the component. 
For this case `export default function SwapHistory(...`

**Review by ledes — APPROVED** [review]

File: src/components/SwapHistory/index.tsx (hunk 7 above)
Comment (hosainnet) [reply src/components/SwapHistory/index.tsx:15]: @ledes is it okay to update in the next PR?

Comment (hosainnet) [reply src/components/SwapHistory/index.tsx]: Done 👍

File: src/utils/token.test.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import constants from "../constants";
+import { getLaunchToken, isLaunchToken } from "utils/token";
+
+test("WETH, USDC and DAI are fund tokens", () => {
```
Comment (pygeek) [inline src/utils/token.test.ts:4]: will approve this, but in future try to keep to one expect per test.

File: src/utils/price.ts [hunk 9]
```diff
@@ -12,6 +13,20 @@ export function mapPrices(tokenPrices: TokenPriceEntry[]): TokenToPrice {
   );
 }
 
+export function convertCurrency(
```
Comment (pygeek) [inline src/utils/price.ts:16]: test this

**Review by pygeek — APPROVED** [review]
Approved and merging, we can address my comments in a separate ticket if necessary.

**Pull request merged**